ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java

```diff
@@ -508,6 +519,122 @@ public void handleDeleteRequest(NetworkRequest request) throws IOException, Inte
+    BatchDeleteRequest batchDeleteRequest;
+    if (request instanceof LocalChannelRequest) {
+      // This is a case where handleDeleteRequest is called when frontends are talking to Azure. In this case, this method
+      // is called by request handler threads running within the frontend router itself. So, the request can be directly
+      // referenced as java objects without any need for deserialization.
+      batchDeleteRequest = (BatchDeleteRequest) ((LocalChannelRequest) request).getRequestInfo().getRequest();
+    } else {
+      batchDeleteRequest = BatchDeleteRequest.readFrom(new DataInputStream(request.getInputStream()), clusterMap);
+    }
+    long requestQueueTime = SystemTime.getInstance().milliseconds() - request.getStartTimeInMs();
+    long totalTimeSpent = requestQueueTime;
+    long startTime = SystemTime.getInstance().milliseconds();
+    boolean operationSuccessful = true;
+    ServerErrorCode operationServerErrorCode = ServerErrorCode.No_Error;
+    List<BatchDeletePartitionResponseInfo> partitionResponseInfoList = new ArrayList<>();
+    BatchDeleteResponse response = new BatchDeleteResponse(batchDeleteRequest.getCorrelationId(), batchDeleteRequest.getClientId(), new ArrayList<>(), ServerErrorCode.No_Error);
+    for (BatchDeletePartitionRequestInfo batchDeletePartitionRequestInfo : batchDeleteRequest.getPartitionRequestInfoList()) {
+      BatchDeletePartitionResponseInfo partitionResponseInfo = null;
+      try {
+        List<StoreKey> convertedStoreKeys = getConvertedStoreKeys(batchDeletePartitionRequestInfo.getBlobIds());
+        ServerErrorCode error =
+            validateRequest(batchDeletePartitionRequestInfo.getPartition(), RequestOrResponseType.BatchDeleteRequest,
+                false);
+        if (error != ServerErrorCode.No_Error) {
+          logger.error("Validating delete request failed with error {} for request {}", error, batchDeleteRequest);
+          List<BlobDeleteStatus> blobsDeleteStatuses = new ArrayList<>();
+          for (StoreKey storeKey: batchDeletePartitionRequestInfo.getBlobIds()){
+            blobsDeleteStatuses.add(new BlobDeleteStatus((BlobId) storeKey, error));
+          }
+          partitionResponseInfo = new BatchDeletePartitionResponseInfo(batchDeletePartitionRequestInfo.getPartition(), blobsDeleteStatuses);
+        } else {
+          List<MessageInfo> infoList = new ArrayList<>();
+          for (StoreKey convertedStoreKey : convertedStoreKeys) {
+            BlobId convertedBlobId = (BlobId) convertedStoreKey;
+            infoList.add(new MessageInfo.Builder(convertedBlobId, -1, convertedBlobId.getAccountId(),
+                convertedBlobId.getContainerId(), batchDeleteRequest.getDeletionTimeInMs()).isDeleted(true)
+                .lifeVersion(MessageInfo.LIFE_VERSION_FROM_FRONTEND)
+                .build());
+          }
+          Store storeToDelete = storeManager.getStore(batchDeletePartitionRequestInfo.getPartition());
```

Comment:
`storeToDelete` gives a signal that we are deleting this store. Maybe rename this to something like `targetStore`?